pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -689,27 +692,64 @@ func CredentialHelpers(sys *types.SystemContext) ([]string, error) {
 	return config.partialV2.CredentialHelpers, nil
 }
 
-// refMatchesPrefix returns true iff ref,
+// refMatchingSubdomainPrefix returns the length of ref
+// iff ref, which is a registry, repository namespace, repository or image reference (as formatted by
+// reference.Domain(), reference.Named.Name() or reference.Reference.String()
+// — note that this requires the name to start with an explicit hostname!),
+// matches a Registry.Prefix value containing wildcarded subdomains in the
+// format: *.example.com. Wildcards are only accepted at the beginning, so
+// other formats like example.*.com will not work. Wildcarded prefixes also
+// cannot contain port numbers or namespaces in them.
+func refMatchingSubdomainPrefix(ref, prefix string) int {
+	index := strings.Index(ref, prefix[1:])
+	if index == -1 {
+		return -1
+	}
+	index += len(prefix[1:])
+	if index == len(ref) {
+		return index
```

Comment:
How was https://github.com/containers/image/pull/1191#discussion_r607873223 resolved? I see some checks against the wildcard `prefix` containing `/` but nothing against the wildcard matching a `/`: so `*.com` will AFAICS match `example.net/namespace-ends-in.com/foo`.

Reply:
let me double-check that.

Reply:
i have added an additional check for this. Will push it in the next commit. Thanks a lot.

Reply:
https://github.com/containers/image/pull/1191/files#diff-522aafc375c87990da7db36f2d4ec60da3b1120aad6b3d18ab0bcf6f1e1bf5caR714

Reply:
test case at https://github.com/containers/image/pull/1191/files#diff-522aafc375c87990da7db36f2d4ec60da3b1120aad6b3d18ab0bcf6f1e1bf5caR714

Reply:
Looks good!